src/mqtt_client.rs

```diff
@@ -308,7 +315,8 @@ where
                             state.maximum_packet_size.replace(size);
                         }
                         Property::AssignedClientIdentifier(id) => {
-                            state.client_id = String::from(id);
+                            state.client_id = String::from_str(id)
+                                .or(Err(Error::ProvidedClientIdTooLong))?;
```

Comment:
What happens if the broker rejects the client_id? Does it sanitize and then emit `AssignedClientIdentifier` in any case?

Reply:
The standard does not indicate required behavior, but it suggests that the broker should reject the message and provide a reason code. In any case, I doubt it would sanitize the ID, but it might, and if it does, we would handle that here.

Reply:
ACK. Handling that case doesn't look critical to me.